quantuminspire/util/configuration.py

```diff
@@ -0,0 +1,72 @@
+"""Module containing the handler for the Quantum Inspire persistent configuration."""
+
+import json
+from pathlib import Path
+from typing import Any, Optional
+
+from pydantic import BaseSettings
+
+
+def ensure_config_file_exists(file_path: Path, file_encoding: Optional[str]) -> None:
+    """Create the file if it does not exist.
+
+    Args:
+        file_path: the file path.
+        file_encoding: The encoding of the file.
+    """
+    if not file_path.exists():
+        file_path.parent.mkdir(parents=True, exist_ok=True)
+        file_path.open("w", encoding=file_encoding).write("{}")
+
+
+def json_config_settings(settings: BaseSettings) -> Any:
+    """Settings source that loads variables from a JSON file specified in the Config class.
+
+    Args:
+        settings: The settings class, used to determine the file encoding, based on the encoding of the
+                  env_file_encoding
+    Returns:
+        A dictionary with the setting variables from the JSON file.
+    """
+    encoding: Optional[str] = settings.Config.env_file_encoding
+    json_config_file: Path = settings.Config.json_config_file  # type: ignore
```

Comment:
The type ignore does not appear necessary here, if it is we should add a specific error to the ignore statement (`#type: ignore[type-of-error]`)

Reply:
pytest is not happy because it cannot "see" that settings.Config has an attribute json_config_file.
Changed the ignore statement into
`# type: ignore[attr-defined]`